datasets/off_category/off_category.py

```diff
@@ -0,0 +1,80 @@
+from collections import namedtuple
+from dataclasses import dataclass
+import gzip
+import orjson
+from typing import Any, Dict
+
+import tensorflow as tf
+import tensorflow_datasets as tfds
+
+
+@dataclass
+class Feature:
+  spec: tfds.features.FeatureConnector
+  default_value: Any
+  input_field: str = None  # use feature name if None
+
+
+_DESCRIPTION = """
+Open Food Facts product category classification dataset.
+"""
+
+_RELEASE_NOTES = {
+  '0.0.1': 'Initial test'
+}
+
+# Don't forget to run `tfds build --register_checksums` when changing the data source
+_DATA_URL = 'https://github.com/openfoodfacts/openfoodfacts-ai/releases/download/dataset-category-2021-09-15/category_xx.{split}.jsonl.gz'
+
+_FEATURES = {
+  'code': Feature(tfds.features.Tensor(shape=(), dtype=tf.string), ''),
+  'product_name': Feature(tfds.features.Tensor(shape=(), dtype=tf.string), ''),
+  'ingredients_tags': Feature(tfds.features.Tensor(shape=(None,), dtype=tf.string), [],
+                              input_field='known_ingredient_tags'),
+  'categories_tags': Feature(tfds.features.Tensor(shape=(None,), dtype=tf.string), [])
+}
+
+_LABEL = 'categories_tags'
+
+
+class OffCategory(tfds.core.GeneratorBasedBuilder):
```

Comment:
Should we call the dataset 'off_category' or 'off_categories'? Repo is 'category-classification' but products have multiple categories. Not sure we care but since tfds manages datasets by name, it'll be more painful to change it later.

Reply:
I think off_categories is better indeed.